6 test cases fail with Truffle test

6 out of 44 test cases fail when running with Truffle.

Run a local Ganache chain:

```sh
$ ganache-cli -p 7545
```

Compile the contracts and test:

```sh
$ yarn compile && yarn truffle test
```

Some test cases fail:

```log
  42 passing (2m)
  6 failing

  1) Contract: LnExchangeSystem
       "before all" hook: prepare suite in "Contract: LnExchangeSystem":
     done() called multiple times in hook <Contract: LnExchangeSystem "before all" hook: prepare suite in "Contract: LnExchangeSystem"> of file ~/repos/linear/linear/test/ExchangeSystem.js; in addition, done() received error: RuntimeError: abort(AssertionError: Operation did not revert as expected: expected false to equal true). Build with -s ASSERTIONS=1 for more info.
    at process.abort (~/.config/truffle/compilers/node_modules/soljson-v0.6.12+commit.27d51765.js:1:13938)
    at process.emit (events.js:327:22)
    at process.emit (~/repos/linear/linear/node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
    at process.emit (~/repos/linear/linear/node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
    at processPromiseRejections (internal/process/promises.js:245:33)
    at processTicksAndRejections (internal/process/task_queues.js:94:32) {
  uncaught: true
}
  Error: done() called multiple times in hook <Contract: LnExchangeSystem "before all" hook: prepare suite in "Contract: LnExchangeSystem"> of file ~/repos/linear/linear/test/ExchangeSystem.js; in addition, done() received error: RuntimeError: abort(AssertionError: Operation did not revert as expected: expected false to equal true). Build with -s ASSERTIONS=1 for more info.
      at process.abort (~/.config/truffle/compilers/node_modules/soljson-v0.6.12+commit.27d51765.js:1:13938)
      at process.emit (node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
      at process.emit (node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
      at processPromiseRejections (internal/process/promises.js:245:33)
      at processTicksAndRejections (internal/process/task_queues.js:94:32) {
    uncaught: true
  }
      at process.emit (node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
      at process.emit (node_modules/truffle/build/webpack:/node_modules/source-map-support/source-map-support.js:495:1)
      at process._fatalException (internal/process/execution.js:156:25)

  2) Contract: test LnBuildBurnSystem
       BuildBurn test:
     Error: Returned error: VM Exception while processing transaction: revert LnExchangeSystem address not valid -- Reason given: LnExchangeSystem address not valid.
      at InitComment (test/common.js:114:24)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Context.<anonymous> (test/TestBuildBurnSystem.js:54:29)

  3) Contract: test LnBuildBurnSystem
       test build burn toTarget:
     Error: Returned error: VM Exception while processing transaction: revert LnExchangeSystem address not valid -- Reason given: LnExchangeSystem address not valid.
      at InitComment (test/common.js:114:24)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Context.<anonymous> (test/TestBuildBurnSystem.js:397:29)

  4) Contract: test LnCollateralSystem
       collateral and redeem:
     Error: Returned error: VM Exception while processing transaction: revert LnExchangeSystem address not valid -- Reason given: LnExchangeSystem address not valid.
      at InitComment (test/common.js:114:24)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Context.<anonymous> (test/TestCollateralSystem.js:32:29)

  5) Contract: test LnCollateralSystem
       IsSatisfyTargetRatio kLnRewardLocker:
     Error: Returned error: VM Exception while processing transaction: revert LnExchangeSystem address not valid -- Reason given: LnExchangeSystem address not valid.
      at InitComment (test/common.js:114:24)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Context.<anonymous> (test/TestCollateralSystem.js:204:29)

  6) Contract: test LnDebtSystem
       delete array to close FeePeriod:
     Error: Returned error: VM Exception while processing transaction: revert LnExchangeSystem address not valid -- Reason given: LnExchangeSystem address not valid.
      at InitComment (test/common.js:114:24)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at Context.<anonymous> (test/TestDebtSystem.js:24:29)

```

--- a/test/DefaultPrices.js
+++ b/test/DefaultPrices.js
@@ -91,7 +91,7 @@ contract('LnDefaultPrices', async (accounts)=> {
             let timeSent = await currentTime();
 
             // should ignore updates not from oracle
-            assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,4 ], timeSent, { from: admin} ) );
+            await assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,4 ], timeSent, { from: admin} ) );
         });
         it('not lusd', async ()=> {
             // new instance of LnDefaultPrices
@@ -99,7 +99,7 @@ contract('LnDefaultPrices', async (accounts)=> {
 
             let timeSent = await currentTime();
 
-            assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("lUSD")], [ 3,4 ], timeSent, { from: oracle} ) );
+            await assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("lUSD")], [ 3,4 ], timeSent, { from: oracle} ) );
         });
         it('not 0', async ()=> {
             // new instance of LnDefaultPrices
@@ -107,7 +107,7 @@ contract('LnDefaultPrices', async (accounts)=> {
 
             let timeSent = await currentTime();
 
-            assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,0 ], timeSent, { from: oracle} ) );
+            await assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,0 ], timeSent, { from: oracle} ) );
         });
 
         it('get price and update time', async () => {
@@ -250,7 +250,7 @@ contract('LnDefaultPrices', async (accounts)=> {
             assert.equal( linaPrice.valueOf(), 0 );
 
             // only oracle
-            assertRevert( defaultPrices.deletePrice( toBytes32("CNY"),{ from:admin} ) );
+            await assertRevert( defaultPrices.deletePrice( toBytes32("CNY"),{ from:admin} ) );
             let cnyPrice = await defaultPrices.getPrice( toBytes32("CNY") );    
             assert.equal( cnyPrice.valueOf(), 2 );
         });
@@ -279,11 +279,11 @@ contract('LnDefaultPrices', async (accounts)=> {
             // new instance of LnDefaultPrices
             const defaultPrices = await LnDefaultPrices.new( admin, oracle, [toBytes32("LINA"), toBytes32("CNY")], [ 1,2 ] );
             // should revert, wrong admin
-            assertRevert( defaultPrices.setOracle( admin, {from:oracle}) );
+            await assertRevert( defaultPrices.setOracle( admin, {from:oracle}) );
 
             // shuld revert, wrong oracle
             let timeSent = await currentTime();
-            assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,4 ], timeSent, { from: oracle } ) );
+            await assertRevert( defaultPrices.updateAll( [toBytes32("LINA"), toBytes32("CNY")], [ 3,4 ], timeSent, { from: admin } ) );
         });
 
 

--- a/test/LnFundVault.js
+++ b/test/LnFundVault.js
@@ -51,7 +51,7 @@ const assertRevert = async (blockOrPromise, reason) => {
 };
 
 
-contract('LnExchangeSystem', async (accounts)=> {
+contract('LnFundVault', async (accounts)=> {
 
     const admin = accounts[0];
     const op1 = accounts[1];
@@ -77,7 +77,7 @@ contract('LnExchangeSystem', async (accounts)=> {
             assert.equal( vaultBalance, v );
 
             // send 1 again
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
 
             // send2
             message = {from: op1, to:vault.address, value: v};
@@ -87,11 +87,11 @@ contract('LnExchangeSystem', async (accounts)=> {
             assert.equal( vaultBalance, v*2 );
 
             // send2 again
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
           
             // send3
             message = {from: op2, to:vault.address, value: v};
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
              
             // claim
             vaultBalance = await web3.eth.getBalance(vault.address);
@@ -123,7 +123,7 @@ contract('LnExchangeSystem', async (accounts)=> {
             assert.equal( vaultBalance, v );
 
             // send 1 again
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
 
             // send2
             message = {from: op1, to:vault.address, value: v};
@@ -133,11 +133,11 @@ contract('LnExchangeSystem', async (accounts)=> {
             assert.equal( vaultBalance, v*2 );
 
             // send2 again
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
           
             // send3
             message = {from: op2, to:vault.address, value: v};
-            assertRevert( web3.eth.sendTransaction(message ) );
+            await assertRevert( web3.eth.sendTransaction(message ) );
              
             // claim
             vaultBalance = await web3.eth.getBalance(vault.address);

--- a/test/TestBuildBurnSystem.js
+++ b/test/TestBuildBurnSystem.js
@@ -49,7 +49,6 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
     it('BuildBurn test', async ()=> {
         const linaBytes32 = toBytes32("lina");
         const ETHBytes32 = toBytes32("ETH");
-        const lusdBytes32 = toBytes32("lUSD");
 
         let InitContracts = await InitComment(ac0);
 
@@ -68,7 +67,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
             "Build amount too big, you need more collateral");
         
         // set price
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(1), toUnit(200), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(1), toUnit(200)], Math.floor(Date.now()/1000).toString() );
     
         // mint lina
         lina.mint(ac1, toUnit(1000) ); // ac1 mint lina
@@ -261,7 +260,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
 
         // price change, debt change, redeem change
         // collateral price down
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(1), toUnit(100), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(1), toUnit(100)], Math.floor(Date.now()/1000).toString() );
         // ac1 collateral 750 lina, value 750 usd
         // ac2 collateral 900 lina, value 900 usd
         // ac3 collateral 5 eth, value 500 usd
@@ -289,7 +288,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
         assert.equal(v, toUnit(0).toString());
 
         // collateral price down down
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(0.1), toUnit(10), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(0.1), toUnit(10)], Math.floor(Date.now()/1000).toString() );
 
         // ac1 collateral 500 lina, value 50 usd
         // ac2 collateral 500 lina, value 50 usd
@@ -318,7 +317,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
             "Build amount too big, you need more collateral");
 
         // collateral price up
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(10), toUnit(1000), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(10), toUnit(1000)], Math.floor(Date.now()/1000).toString() );
 
         // ac1 collateral 500 lina, value 5000 usd
         // ac2 collateral 500 lina, value 5000 usd
@@ -352,7 +351,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
         assert.equal(v, toUnit(3000).toString());
 
         // price down
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(0.1), toUnit(10), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(0.1), toUnit(10)], Math.floor(Date.now()/1000).toString() );
 
         v = await kLnCollateralSystem.MaxRedeemableInUsd(ac1);
         assert.equal(v, toUnit(0).toString());
@@ -392,7 +391,6 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
     it('test build burn toTarget', async ()=> {
         const linaBytes32 = toBytes32("lina");
         const ETHBytes32 = toBytes32("ETH");
-        const lusdBytes32 = toBytes32("lUSD");
 
         let InitContracts = await InitComment(ac0);
 
@@ -407,7 +405,7 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
         await kLnCollateralSystem.UpdateTokenInfo( linaBytes32, lina.address, toUnit(1), false);
         
         // set price
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit(1), toUnit(200), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit(1), toUnit(200)], Math.floor(Date.now()/1000).toString() );
         
         // mint lina
         lina.mint(ac1, toUnit(1000) ); // ac1 mint lina
@@ -426,12 +424,12 @@ contract('test LnBuildBurnSystem', async (accounts)=> {
 
         checkDebtBalance(ac1, toUnit(200));
 
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit("0.5"), toUnit(200), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit("0.5"), toUnit(200)], Math.floor(Date.now()/1000).toString() );
         await kLnBuildBurnSystem.BurnAssetToTarget({from:ac1});
 
         checkDebtBalance(ac1, toUnit(100));
 
-        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32, lusdBytes32], [toUnit("1"), toUnit(200), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32, ETHBytes32], [toUnit("1"), toUnit(200)], Math.floor(Date.now()/1000).toString() );
         await exceptionEqual(
             kLnBuildBurnSystem.BurnAssetToTarget({from:ac1}),
             "You maybe want build to target"

--- a/test/TestCollateralSystem.js
+++ b/test/TestCollateralSystem.js
@@ -25,7 +25,6 @@ contract('test LnCollateralSystem', async (accounts)=> {
 
     const linaBytes32 = toBytes32("lina");
     const ETHBytes32 = toBytes32("ETH");
-    const lusdBytes32 = toBytes32("lUSD");
 
     it('collateral and redeem', async ()=> {
         
@@ -212,7 +211,7 @@ contract('test LnCollateralSystem', async (accounts)=> {
         let lUSD = InitContracts.lUSD;
 
         await kLnCollateralSystem.UpdateTokenInfo( linaBytes32, lina.address, toUnit(1), false);
-        await kLnChainLinkPrices.updateAll([linaBytes32, lusdBytes32], [toUnit(1), toUnit(1)], Math.floor(Date.now()/1000).toString() );
+        await kLnChainLinkPrices.updateAll([linaBytes32], [toUnit(1)], Math.floor(Date.now()/1000).toString() );
 
         let v = await kLnCollateralSystem.GetSystemTotalCollateralInUsd();
         assert.equal(v.valueOf(), 0);

--- a/test/common.js
+++ b/test/common.js
@@ -101,6 +101,7 @@ async function InitComment(admin) {
     registContract("LnBuildBurnSystem", kLnBuildBurnSystem);
     registContract("LnFeeSystem", kLnFeeSystem);
     registContract("LnRewardLocker", kLnRewardLocker);
+    registContract("LnExchangeSystem", kLnExchangeSystem);
   
     await kLnAssetSystem.updateAll(contractNames, contractAddrs);
 

